acclimatise/model.py

```diff
@@ -371,33 +371,62 @@ def capital(self):
         )
 
 
-int_re = re.compile("(int(eger)?)|size|length|max|min", flags=re.IGNORECASE)
-str_re = re.compile("str(ing)?", flags=re.IGNORECASE)
-float_re = re.compile("float|decimal", flags=re.IGNORECASE)
-bool_re = re.compile("bool(ean)?", flags=re.IGNORECASE)
-file_re = re.compile("file|path", flags=re.IGNORECASE)
-dir_re = re.compile("folder|directory", flags=re.IGNORECASE)
+int_re = re.compile(
+    r"\b((int(eger)?)|size|length|max|min|(num(ber)?))\b", flags=re.IGNORECASE
+)
+str_re = re.compile(r"\bstr(ing)?\b", flags=re.IGNORECASE)
+float_re = re.compile(r"\b(float|decimal)\b", flags=re.IGNORECASE)
+bool_re = re.compile(r"\bbool(ean)?\b", flags=re.IGNORECASE)
+file_re = re.compile(r"\b(file|path)\b", flags=re.IGNORECASE)
+input_re = re.compile(r"input", flags=re.IGNORECASE)
+output_re = re.compile(r"output", flags=re.IGNORECASE)
+dir_re = re.compile(r"\b(folder|directory)\b", flags=re.IGNORECASE)
+
+default_re = re.compile(
+    r"default(?: value)?(?:[:=] ?| )([^ )\]]+)", flags=re.IGNORECASE
```

Comment:
If I understand correctly what you're doing here, `(?: value)?` could be simplified to `( value)*`.
But also, would you mind briefly explaining this and the below regex with a comment? The others are self explanatory I think

Reply:
From the docs: 

```
(?:...)

    A non-capturing version of regular parentheses. Matches whatever regular expression is inside the parentheses, but the substring matched by the group cannot be retrieved after performing a match or referenced later in the pattern.
````

Just doing this to be able to get the content of the default value by accessing `.group(1)` irrespective of the fact if `( value)` was included in the string or not.

Reply:
Oh, yep I misunderstood. That makes sense.

Reply:
Although, how come this `default` stuff is in this branch and not isolated to #16?

Reply:
Will move it.